terraform/modules/common/instance/main.tf

```diff
@@ -0,0 +1,63 @@
+## Variables
+
+variable "stack_name" {
+  description = "The name of the stack within which the resource resides"
+}
+
+variable "ami_id" {
+  description = "The ami which will be used to build the instance"
+}
+
+variable "subnet_id" {
+  description = "The subnet which the instance will be added to"
+}
+
+variable "ssh_key_name" {
+  description = "The name of the ssh key which will assigned to the instance"
+}
+
+variable "security_groups" {
+  description = "A list of security groups which will be added to the instance. If the list is empty the default securit group will be used"
+  type        = "list"
+}
+
+variable "public_ip" {
+  description = "When this is true a public ip is added to the instance"
+  default     = false
+}
+
+## Resources
+
+resource "aws_instance" "instance" {
+  ami           = "${var.ami_id}"
+  subnet_id     = "${var.subnet_id}"
+  instance_type = "t2.small"
```

Comment:
Instance type could be put in a variable with t2.small as the default as it seems a bit specific to the jumpbox (I don't mind if you don't address this comment too much by the way)

Reply:
Think this module should be part of jumpbox project